### PR TITLE
feat: add to contacts button for peer chat #778

### PR DESCRIPTION
**Pull request by FrolYo**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor [hunk 1]
```diff
@@ -43,15 +57,44 @@
     [Inject] private ModalUI ModalUI { get; init; } = null!;
     [Inject] private RightPanelUI RightPanelUI { get; init; } = null!;
     [Inject] private NavbarUI NavbarUI { get; init; } = null!;
+    [Inject] private UICommander UICommander { get; init; } = null!;
+    [Inject] private Session Session { get; init; } = null!;
```
Comment (alexyakunin) [inline src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor:61]: @FrolYo let's use "standard" order of injections - Session, replica/compute services, UI services, etc.

File: src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor [hunk 2]
```diff
@@ -43,15 +57,44 @@
...
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
+        if (Chat.Kind == ChatKind.Peer) {
+            var contact = await Contacts.GetForChat(Session, Chat.Id, cancellationToken);
+            return new(!contact.IsStored(), false, contact);
+        }
+
+        return new(false, Chat.Rules.CanInvite(), null);
+    }
+
+    public sealed record Model(bool CanAddToContacts, bool CanInvite, Contact? Contact) {
```
Comment (alexyakunin) [inline src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor:97]: Let's make `CanAddToContacts` as computed property based on `Contact`.

**Review by alexyakunin — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor (hunk 2 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor]: done

File: src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor (hunk 1 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor:59]: done

**Pull request merged**